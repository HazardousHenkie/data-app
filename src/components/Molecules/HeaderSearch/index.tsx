import React, { useState } from 'react'

import CountriesList from 'components/Molecules/CountriesList'
import SearchField from 'components/Molecules/SearchField'

import TopBar, {
    CloseCountriesList,
    SearchFieldWrapper
} from './styledComponents'

const HeaderSearch: React.FC = () => {
    const [searchString, setSearchString] = useState('')
    const [openCountriesList, setOpenCountriesList] = React.useState(false)

    const closeCountriesListClick = () => {
        setOpenCountriesList(false)
    }

    const openCountriesListClick = () => {
        setOpenCountriesList(true)
    }

    return (
        <>
            <TopBar>
                <CloseCountriesList onClick={closeCountriesListClick} />
                <SearchFieldWrapper
                    role="button"
                    tabIndex={0}
                    onClick={openCountriesListClick}
                    onKeyDown={openCountriesListClick}
                >
                    <SearchField setValue={setSearchString} />
                </SearchFieldWrapper>
            </TopBar>
            <CountriesList
                open={openCountriesList}
                searchString={searchString}
            />
        </>
    )
}

export default HeaderSearch
