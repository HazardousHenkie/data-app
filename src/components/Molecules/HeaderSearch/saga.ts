import {
    call,
    put,
    takeLatest,
    ForkEffect,
    PutEffect,
    CallEffect
} from 'redux-saga/effects'

import request from 'utils/request'

// eslint-disable-next-line
import { getCountriesDataSuccess, getCountriesDataError } from './actions' // eslint-disable-line 
import ActionTypes from './constants' // eslint-disable-line 

function* getCountriesDataSaga(): Generator<
    | CallEffect<unknown>
    | PutEffect<getCountriesDataSuccess | getCountriesDataError>,
    void,
    unknown
> {
    try {
        const requestURL = 'https://restcountries.eu/rest/v2/all'

        yield call(request, requestURL, {
            method: 'GET'
        })

        yield put(getCountriesDataSuccess())
    } catch (error) {
        yield put(getCountriesDataError(error))
    }
}

export default function* getCountriesDataRootSaga(): Generator<
    ForkEffect<never>,
    void,
    unknown
> {
    yield takeLatest(ActionTypes.GET_COUNTRIES_DATA, getCountriesDataSaga)
}
