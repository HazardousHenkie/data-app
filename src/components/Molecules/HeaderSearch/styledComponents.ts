import styled from 'styled-components'

import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'

const TopBar = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1350;
    padding: 15px;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`

export const CloseCountriesList = styled(ChevronLeftIcon)`
    font-size: 3rem;
    cursor: pointer;
    color: ${(props): string => props.theme.black};
`

export default TopBar
