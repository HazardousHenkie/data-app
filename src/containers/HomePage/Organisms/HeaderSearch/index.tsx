import React, { useState } from 'react'

import CountriesList from 'containers/HomePage/Molecules/CountriesList'
import SearchField from 'components/Molecules/SearchField'
import SideDrawer from 'components/Organisms/SideDrawer'
import MenuList from '../../Molecules/MenuList'

import TopBar, {
    CloseCountriesList,
    SearchFieldWrapper
} from './styledComponents'

const HeaderSearch: React.FC = () => {
    const [searchString, setSearchString] = useState('')
    const [openCountriesList, setOpenCountriesList] = React.useState(false)

    const closeCountriesListClick = () => {
        setOpenCountriesList(false)
    }

    const openCountriesListClick = () => {
        setOpenCountriesList(true)
    }

    return (
        <>
            <TopBar data-testid="TopBar">
                {!openCountriesList && (
                    <SideDrawer>
                        <MenuList />
                    </SideDrawer>
                )}
                {openCountriesList && (
                    <CloseCountriesList
                        data-testid="CloseCountriesList"
                        onClick={closeCountriesListClick}
                    />
                )}
                <SearchFieldWrapper
                    data-testid="SearchFieldWrapper"
                    role="button"
                    tabIndex={0}
                    onClick={openCountriesListClick}
                    onKeyDown={openCountriesListClick}
                >
                    <SearchField setValue={setSearchString} />
                </SearchFieldWrapper>
            </TopBar>

            <CountriesList
                open={openCountriesList}
                setOpen={setOpenCountriesList}
                searchString={searchString}
            />
        </>
    )
}

export default HeaderSearch
