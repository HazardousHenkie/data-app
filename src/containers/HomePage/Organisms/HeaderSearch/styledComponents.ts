import styled, { keyframes } from 'styled-components'

import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'

const TopBar = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1350;
    padding: 0 10px;
    margin: 15px;
    width: calc(100% - 30px);
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 10px;
    box-shadow: ${props => props.theme.shadow};
    background: ${({ theme }) => theme.palette.primary.main};
`

export const SearchFieldWrapper = styled.div`
    width: 100%;
`

export const RotateNinetyDegrees = keyframes`
    0% {
        transform: rotate(90deg);
    }
    100% {
        transform: rotate(0deg);
    }
`

export const CloseCountriesList = styled(ChevronLeftIcon)`
    font-size: 2.5rem;
    width: 35px;
    margin-right: 7px;
    cursor: pointer;
    animation: ${RotateNinetyDegrees} 0.4s ease-in-out;
    color: ${({ theme }) => theme.palette.common.black};
`

export default TopBar
