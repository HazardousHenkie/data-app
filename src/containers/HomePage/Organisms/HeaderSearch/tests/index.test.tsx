import React from 'react'
import renderer from 'react-test-renderer'

import MockingAppComponent from 'utils/testing/MockingAppComponent'
import HeaderSearch from '../index'

describe('<HeaderSearch />', () => {
    it('should render like snapshot', () => {
        const component = renderer.create(
            <MockingAppComponent>
                <HeaderSearch />
            </MockingAppComponent>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
