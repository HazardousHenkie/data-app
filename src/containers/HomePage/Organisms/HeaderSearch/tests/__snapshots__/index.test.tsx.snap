// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderSearch /> should render like snapshot 1`] = `
Array [
  <div
    className="sc-fznyAO ktFYgq"
  >
    <button
      aria-label="open drawer"
      className="MuiButtonBase-root MuiIconButton-root sc-AxheI eWObgQ MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <div
      className="sc-fznKkj cHSvOt"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <form />
    </div>
  </div>,
  <div
    className="sc-AxirZ ixIRlM"
    style={
      Object {
        "opacity": 0,
        "visibility": "hidden",
      }
    }
  >
    <nav
      aria-label="countries"
      className="MuiList-root MuiList-padding"
    />
  </div>,
]
`;
